From d4305208955c5cdd9fe96dfa61e7c1e14e176a14 Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Sun, 17 Jan 2021 11:05:36 -0800
Subject: [PATCH] Add group terminator if it is left off.

If there is no terminating ";" add one to the list, to make the text
re-rendering correct.
---
 rfc822.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/rfc822.c b/rfc822.c
index 6a375ba8..2f56e070 100644
--- a/rfc822.c
+++ b/rfc822.c
@@ -656,6 +656,10 @@ ADDRESS *rfc822_parse_adrlist (ADDRESS *top, const char *s)
     last->val = mutt_substrdup (begin, s - nl < begin ? begin : s - nl);
 #endif
 
+  /* add group terminator, if it was left off */
+  if (last && in_group)
+    last->next = rfc822_new_address ();
+
   return top;
 }
 
-- 
GitLab

